src/java/org/apache/cassandra/db/compaction/UnifiedCompactionStrategy.java

```diff
@@ -149,12 +150,27 @@ public void shutdown()
     @Override
     public synchronized Collection<AbstractCompactionTask> getNextBackgroundTasks(int gcBefore)
     {
+        // TODO - we should perhaps consider executing this code less frequently than legacy strategies
+        // since it's more expensive, and we should therefore prevent a second concurrent thread from executing at all
```

Comment:
I am not certain, but I think maybe not because this method `getNextBackgroundTask(int)` is used in CC itself and to my understanding this is what the comment refers to. otoh, `getNextBackgroundTask(Collection<CpmpactionAggregate>, int)` is only directly used from cndb.